examples/python/ocr/ocr.py

```diff
@@ -299,56 +314,101 @@ def update_zoom_paths(
-    paths, detections_paths, zoom_paths_figures, zoom_paths_tables, zoom_paths_texts = process_layout_records(layout)
+def generate_blueprint(
+    layouts: list[Layout],
+    processed_layouts: list[LayoutStructure],
+) -> rrb.Blueprint:
+    page_tabs = []
+    for layout, processed_layout in zip(layouts, processed_layouts):
+        paths, detections_paths, zoom_paths_figures, zoom_paths_tables, zoom_paths_texts = processed_layout
 
-    tabs = []
-    content_data: dict[str, Any] = {
-        "Figures": zoom_paths_figures,
-        "Tables": zoom_paths_tables,
-        "Texts": zoom_paths_texts,
-    }
+        section_tabs = []
+        content_data: dict[str, Any] = {
+            "Figures": zoom_paths_figures,
+            "Tables": zoom_paths_tables,
+            "Texts": zoom_paths_texts,
+        }
 
-    for name, paths in content_data.items():
-        if paths:
-            tabs.append(rrb.Tabs(*paths, name=name))  # type: ignore[arg-type]
+        for name, paths in content_data.items():
+            if paths:
+                section_tabs.append(rrb.Tabs(*paths, name=name))  # type: ignore[arg-type]
+
+        page_path = f"page_{layout.page_number}"
+        page_tabs.append(
+            rrb.Vertical(
+                rrb.Horizontal(
+                    rrb.Spatial2DView(
+                        name="Layout",
+                        origin=f"{page_path}/Image/",
+                        contents=[f"{page_path}/Image/**"] + detections_paths,
+                    ),
+                    rrb.Spatial2DView(name="Detections", contents=[f"{page_path}/Image/**"]),
+                    rrb.TextDocumentView(name="Recovery", contents=f"{page_path}/Recovery"),
+                ),
+                rrb.Horizontal(*section_tabs),
+                name=page_path,
+                row_shares=[4, 3],
+            )
+        )
 
     return rrb.Blueprint(
-        rrb.Vertical(
-            rrb.Horizontal(
-                rrb.Spatial2DView(name="Layout", origin="Image/", contents=["Image/**"] + detections_paths),
-                rrb.Spatial2DView(name="Detections", contents=["Image/**"]),
-                rrb.TextDocumentView(name="Recovery", contents="Recovery"),
-            ),
-            rrb.Horizontal(*tabs),
-            row_shares=[4, 3],
-        ),
+        rrb.Tabs(*page_tabs),
         collapse_panels=True,
     )
 
 
-def detect_and_log_layout(img_path: str) -> None:
-    # Layout Object - This will contain the detected layouts and their detections
-    layout = Layout()
+def detect_and_log_layouts(file_path: str) -> None:
+    images: list[npt.NDArray[np.uint8]] = []
+    if file_path.endswith(".pdf"):
+        # convert pdf to images
+        images.extend(np.array(img, dtype=np.uint8) for img in pdf2image.convert_from_path(file_path))
+    else:
+        # read image
+        img = cv2.imread(file_path)
+        coloured_image = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
+        images.append(coloured_image.astype(np.uint8))
```

Comment:
For consistency, we use American English spelling in this repository

```suggestion
        colored_image = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
        images.append(colored_image.astype(np.uint8))
```

…though `image_rgb` would be an even better name imho

Reply:
Changed all the variables called `coloured_image` to `image_rgb`